minio/minio
# Fix SSE-C checksums

## Description

SSE-C checksums could not be decrypted, since it missed the key.

Should also s3zip files with SSE-C.

## How to test this PR?
```
λ aws s3api --debug --endpoint-url https://127.0.0.1:9001 put-object --bucket testbucket --key hello.txt --body hello.txt --sse-customer-algorithm AES256 --sse-customer-key "2teLYwr0HB84FYUxvhiTG//u3C4/RnI2qhRt6jn64K0=" --sse-customer-key-md5 "h06pENAWWRBnqFm+lHPtUA==" --no-verify-ssl --checksum-algorithm SHA256 --checksum-sha256 CKw2JCTHBUK0z6+wVutMXRw4SDyMxn2eIEHQDu3o8d4=
{
    "ETag": "\"504bad1096d7cc8817127db0e25afc56\"",
    "ChecksumSHA256": "CKw2JCTHBUK0z6+wVutMXRw4SDyMxn2eIEHQDu3o8d4=",
    "VersionId": "f6805362-a947-4f99-9096-e67cb47db11e",
    "SSECustomerAlgorithm": "AES256",
    "SSECustomerKeyMD5": "h06pENAWWRBnqFm+lHPtUA=="
}

λ aws s3api head-object --bucket testbucket --key hello.txt --endpoint-url https://localhost:9001 --checksum-mode ENABLED  --no-verify-ssl --sse-customer-algorithm AES256 --sse-customer-key "2teLYwr0HB84FYUxvhiTG//u3C4/RnI2qhRt6jn64K0=" --sse-customer-key-md5 "h06pENAWWRBnqFm+lHPtUA=="
{
    "AcceptRanges": "bytes",
    "LastModified": "2024-06-07T09:08:51+00:00",
    "ContentLength": 8,
    "ChecksumSHA256": "CKw2JCTHBUK0z6+wVutMXRw4SDyMxn2eIEHQDu3o8d4=",
    "ETag": "\"504bad1096d7cc8817127db0e25afc56\"",
    "VersionId": "f6805362-a947-4f99-9096-e67cb47db11e",
    "ContentType": "binary/octet-stream",
    "Metadata": {},
    "SSECustomerAlgorithm": "AES256",
    "SSECustomerKeyMD5": "h06pENAWWRBnqFm+lHPtUA==",
}

λ aws s3api get-object --bucket testbucket --key hello.txt --endpoint-url https://localhost:9001 --checksum-mode ENABLED  --no-verify-ssl --sse-customer-algorithm AES256 --sse-customer-key "2teLYwr0HB84FYUxvhiTG//u3C4/RnI2qhRt6jn64K0=" --sse-customer-key-md5 "h06pENAWWRBnqFm+lHPtUA==" out.txt
{
    "AcceptRanges": "bytes",
    "LastModified": "2024-06-07T09:08:51+00:00",
    "ContentLength": 8,
    "ETag": "\"504bad1096d7cc8817127db0e25afc56\"",
    "ChecksumSHA256": "CKw2JCTHBUK0z6+wVutMXRw4SDyMxn2eIEHQDu3o8d4=",
    "VersionId": "f6805362-a947-4f99-9096-e67cb47db11e",
    "ContentType": "binary/octet-stream",
    "Metadata": {},
    "SSECustomerAlgorithm": "AES256",
    "SSECustomerKeyMD5": "h06pENAWWRBnqFm+lHPtUA==",
}
```

Works with already uploaded objects.

## Types of changes
